### ISSUE TITLE
Compiler doesn't throw error when base class with Rest parameters is incorrectly extended.

### ISSUE THREAD

**Issue by Izhaki** [Microsoft/TypeScript#12929]
**TypeScript Version:**  2.0.3

**Code**

Note the different parameters of `addViewees`:

```ts
abstract class ElementLayer {
    addViewees( ...aViewees: Viewee[] ): void {
         ...
    }
}

class Layer extends ElementLayer {
    addViewees( aViewee: Viewee ): void {
        super.addViewees( aViewee );
        ...
    }
}
```

**Expected behavior:**

Compiler to complain:

> Class 'Layer' incorrectly extends base class 'ElementLayer'.

**Actual behavior:**

No error.

**Comment by Izhaki** [comment]
Now if I change the code to this:

```ts
abstract class ElementLayer {
    addViewees( ...aViewees: Viewee[] ): void {
         ...
    }
}

class Layer extends ElementLayer {
    addViewees( ...aViewees: Viewee[] ): void {
        super.addViewees( aViewees );
        ...
    }
}
```

I get:

> Argument of type 'Viewee[]' is not assignable to parameter of type 'Viewee'.

Which, if I'm honest, makes sense.

So I guess the question is really: How do you extend a base class method that takes rest parameters?

**Comment by Izhaki** [comment]
Closing this issue as I can see now why the code in the original code is valid.

**Comment by Izhaki** [comment]
For those who wonder:

```ts
abstract class ElementLayer {
    addViewees( ...aViewees: Viewee[] ): void {
         ...
    }
}

class Layer extends ElementLayer {
    addViewees( ...aViewees: Viewee[] ): void {
        super.addViewees( ...aViewees );
        ...
    }
}
```
